src/reader.rs

```diff
@@ -72,18 +72,20 @@ impl<R: Read> BzDecompressor<R> {
 
 impl<R: Read> Read for BzDecompressor<R> {
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        self.0.read(|stream, input, _eof| {
-            stream.decompress(input, buf)
+        self.0.read(buf.len(), |stream, input, offset, _eof| {
+            stream.decompress(input, &mut buf[offset..])
```

Comment:
Hm so in theory this should never be necessary, any call to `read` will finish immediately after at most one byte of data has been consumed. Is there some sample input I could test out where it fails today? (or perhaps the test cases included are good enough?)